### ISSUE TITLE
[feature request] Expand a given dim

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#62352]
This would be a simpler API for torch.repeat / expand, Or maybe does it exist already?

```python
import torch

expand_dim = lambda tensor, dim, expand: tensor.unsqueeze(dim).expand((-1, ) * (dim if dim >= 0 else tensor.ndim + dim + 1) + (expand, ) + (-1, ) * (tensor.ndim - (dim if dim >= 0 else tensor.ndim + dim + 1)))

a = torch.rand(3, 4, 5)
b = expand_dim(a, dim = -2, expand =  6)
c = expand_dim(a, dim = 3, expand =  8)

print(a.shape)
# torch.Size([3, 4, 5])
print(b.shape)
# torch.Size([3, 4, 6, 5])
print(c.shape)
# torch.Size([3, 4, 5, 8])
```

**Comment by vadimkantorov** [comment]
maybe a kwarg-only `dim` argument could be added  to `expand`, and that would limit its application to provided dim (or even multidim). Also a question of whether it should do unsqueeze or not
